### PR TITLE
Implement create resource group step.

### PR DESCRIPTION
**Pull request by boumenot**
Part of the changes needed for #94

### CODE REVIEW

File: packer/builder/azure/arm/step_create_resource_group.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		state.Put(constants.Error, err)
+		s.error(err)
+
+		return multistep.ActionHalt
+	}
+
+	return multistep.ActionContinue
+}
+
+func (*StepCreateResourceGroup) Cleanup(multistep.StateBag) {
```
Comment (paulmey) [inline packer/builder/azure/arm/step_create_resource_group.go]: You probably want to clean up the resource group in here. (This is also only half-used in the SMAPI version.)

File: packer/builder/azure/arm/step_create_resource_group.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+func NewStepCreateResourceGroup(client *AzureClient, ui packer.Ui) *StepCreateResourceGroup {
+	var step = &StepCreateResourceGroup{
+		client: client,
+		say:    func(message string) { ui.Say(message) },
+		error:  func(e error) { ui.Error(e.Error()) },
+	}
+
+	step.create = step.createResourceGroup
```
Comment (paulmey) [inline packer/builder/azure/arm/step_create_resource_group.go]: These can actually all be part of the struct initialization above...

Comment (boumenot) [inline packer/builder/azure/arm/step_create_resource_group.go]: How?  Isn't that a :chicken: and :egg: problem?

File: packer/builder/azure/arm/step_create_resource_group.go (hunk 1 above)
Comment (boumenot) [inline packer/builder/azure/arm/step_create_resource_group.go]: I've added this to the TODO list.  I need to better understand what and how packer treats cleanup.

File: packer/builder/azure/arm/step_create_resource_group.go (hunk 2 above)
Comment (paulmey) [inline packer/builder/azure/arm/step_create_resource_group.go]: Yes, you're right. I was too fast with my judgement...
BTW, a :egg: will no longer yield a :chicken: (baked like that)

**Pull request merged**